### PR TITLE
Added functions required to support fee delegation transaction

### PR DESCRIPTION
**Pull request by cp-wjhan**

### CODE REVIEW

File: core/error.go [hunk 1]
```diff
@@ -96,4 +96,16 @@ var (
...
 	// ErrSenderNoEOA is returned if the sender of a transaction is a contract.
 	ErrSenderNoEOA = errors.New("sender not an eoa")
+
+	// fee delegation
+	// ErrInvalidFeePayer is returned if the transaction contains an invalid feePayer's signature.
+	ErrInvalidFeePayer = errors.New("invalid fee delegation feePayer")
+
+	// ErrFeePayerInsufficientFunds is returned if the fee cost of executing a transaction
+	// is higher than the balance of the feePayer's account.
+	ErrFeePayerInsufficientFunds = errors.New("fee delegation feePayer's insufficient funds for gas * price")
```
Comment (trident90) [inline core/error.go]: How about "insufficient fee delegation feePayer's funds for gas * price"?

Comment (trident90) [inline core/error.go]: How about "insufficient fee delegation feePayer's funds for value"?

File: core/state_transition.go [hunk 2]
```diff
@@ -194,24 +196,57 @@ func (st *StateTransition) to() common.Address {
...
-		return fmt.Errorf("%w: address %v have %v want %v", ErrInsufficientFunds, st.msg.From().Hex(), have, want)
-	}
-	if err := st.gp.SubGas(st.msg.Gas()); err != nil {
-		return err
-	}
-	st.gas += st.msg.Gas()
+	// fee delegation
+	if st.msg.FeePayer() != nil {
+		if !st.evm.ChainConfig().IsFeeDelegation(st.evm.Context.BlockNumber) {
+			return fmt.Errorf("%w: fee delegation type not supperted", ErrTxTypeNotSupported)
```
Comment (trident90) [inline core/state_transition.go:202]: please correct typo - "supperted"

**Review by trident90 — CHANGES_REQUESTED** [review]

File: core/error.go (hunk 1 above)
Comment (cp-wjhan) [reply core/error.go]: I will change it as follows.
ErrInvalidFeePayer = errors.New("fee delegation: invalid feePayer")
ErrFeePayerInsufficientFunds = errors.New("fee delegation: insufficient feePayer's funds for gas * price")
ErrSenderInsufficientFunds = errors.New("fee delegation:  insufficient sender funds for value")

File: core/state_transition.go (hunk 2 above)
Comment (cp-wjhan) [reply core/state_transition.go:202]: I'll fix it.

**Review by trident90 — APPROVED** [review]
LGTM

**Review by cp-khs — COMMENTED** [review]
LGTM
It will be a great feature to improve the versatility of transactions.

**Pull request merged**